app/controllers/api/v1/scores_controller.rb

```diff
@@ -0,0 +1,120 @@
+class Api::V1::ScoresController < Api::V1::BaseApiController
+
+  before_action -> { require_privilege :instructor_all }
+  before_action :set_assessment
+
+  def index
+    submissions = @assessment.submissions.where(assessment_id: @assessment.id)
+
+    scores = {}
+    cud_to_user = {}
+    submissions.each do |submission|
+      cud = submission.course_user_datum_id
+      unless cud_to_user.key?(cud)
+        cud_to_user[cud] = User.find(CourseUserDatum.find(cud).user_id)
+      end
+      user = cud_to_user[cud]
+
+      raw_submission_scores = Score.where(submission_id: submission.id)
+      user_scores = scores.fetch(user.email, {})
+      user_scores[submission.version] = problem_name_to_score(raw_submission_scores)
+
+      scores[user.email] = user_scores
+    end
+
+    respond_with scores
+  end
+
+  def show
+    cud = get_cud params
+    submissions = @assessment.submissions.where(
+      assessment_id: @assessment.id,
+      course_user_datum_id: cud.id
+    )
+
+    scores = {}
+    submissions.each do |submission|
+      raw_submission_scores = Score.where(submission_id: submission.id)
+      scores[submission.version] = problem_name_to_score(raw_submission_scores)
+    end
+
+    respond_with scores
+  end
+
+  def update
```

Comment:
This is actually a bit misleading of an update endpoint. The `update` endpoint is associated with an id, so in this case, the update should be accepting the score id as a query param and updating that specific score.

I do understand the preferred use case of just updating the latest score like you did however - but in that case do just create a separate endpoint called update_latest perhaps?

Reply:
Will change this to `update_latest`